csi/controller_server.go

```diff
@@ -545,6 +547,40 @@ func (cs *ControllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 		return nil, status.Error(codes.InvalidArgument, "Snapshot name must be provided")
 	}
 
+	volume, err := cs.apiClient.Volume.ById(csiVolumeName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get volume %v", csiVolumeName)
+	}
+	if volume.State == string(longhorn.VolumeStateDetached) {
+		nodeID, err := cs.autoAttachVolumeToRandomNode(volume)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		if nodeID != "" {
```

Comment:
Question: Why can there be an empty node ID with no error returned after the auto attachment?

Reply:
There shouldn't be a case. Nice catch. Fixed.